src/autoload/global.gd

```diff
@@ -113,69 +118,76 @@ func _ready() -> void:
 
 
 func save_config() -> void:
-	var config_file := FileAccess.open(CONFIG_PATH, FileAccess.WRITE)
-	if not config_file:
-		push_error("Could not open config file for writing!")
-		return
+	var config_file := ConfigFile.new()
 
 	if get_window().mode == Window.MODE_FULLSCREEN:
 		config["fullscreen"] = true
 	else:
 		config["fullscreen"] = false
 		config["window_size"] = get_window().size
 
-	config_file.store_var(config)
-	config_file.close()
+	for key in config:
+		config_file.set_value(CONFIG_SECTION, key, config[key])
+
+	var err := config_file.save(CONFIG_PATH)
+	if err != OK:
+		push_error("Could not write to config file!")
 
 
 func load_config() -> void:
-	var config_file := FileAccess.open(CONFIG_PATH, FileAccess.READ)
-	var open_error := FileAccess.get_open_error()
+	var config_file := ConfigFile.new()
+	var open_error := config_file.load(CONFIG_PATH)
 	if open_error == ERR_FILE_NOT_FOUND:
 		print("No config file found, using default settings")
 		return
 	elif open_error:
 		push_warning("Could not open config file for reading! Using default settings")
 		return
 
-	var new_config_variant = config_file.get_var()
-	config_file.close()
-
-	if typeof(new_config_variant) != TYPE_DICTIONARY:
-		push_warning("Config file was corrupted! Using default settings")
+	if not config_file.has_section(CONFIG_SECTION):
+		push_warning("Config missing section %s, using default settings" % CONFIG_SECTION)
 		return
-	var new_config := new_config_variant as Dictionary
-
-	config = new_config.duplicate()
 
-	for entry in DEFAULT_CONFIG:
-		if entry not in config:
-			config[entry] = DEFAULT_CONFIG[entry]
+	for key in config:
+		if config_file.has_section_key(CONFIG_SECTION, key):
+			var value_variant = config_file.get_value(CONFIG_SECTION, key)
+			if typeof(value_variant) == typeof(DEFAULT_CONFIG[key]):
+				if key in ["music_volume", "sound_volume"]:
+					value_variant = clampf(value_variant, 0.0, 1.0)
+				elif key == "anim_key_idx":
+					value_variant = clampi(value_variant, 0, ANIMATION_SPEEDS.size() - 1)
+				config[key] = value_variant
```

Comment:
Honestly since we have custom logic for every single key here, we might as well ignore the for loop and do each individually.

Reply:
Moved the range-checking to after the for loop